cdc_replication_test.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+import shutil
+import sys
+import os
+import time
+from tempfile import TemporaryDirectory
+from textwrap import dedent
+
+from cassandra import ConsistencyLevel
+from cassandra.query import SimpleStatement
+
+from sdcm.tester import ClusterTester
+
+def print_file_to_stdout(path: str) -> None:
+    with open(path, "r") as f:
+       shutil.copyfileobj(f, sys.stdout)
+
+# Write a CQL select result to a file.
+def write_cql_result(res, path: str):
+    with open(path, 'w') as f:
+        for r in res:
+            f.write(str(r) + '\n')
+        f.flush()
+        os.fsync(f.fileno())
+
+SCYLLA_MIGRATE_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-migrate"
+REPLICATOR_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-cdc-replicator-0.0.1-SNAPSHOT-jar-with-dependencies.jar"
+
+def keyspace_def(rf: int) -> str:
+    return f"create keyspace if not exists ks with replication = {{'class': 'SimpleStrategy', 'replication_factor': {rf}}}"
+
+TABLE_DEF_BASE = "create table if not exists ks.tb (pk int, ck int, v text, primary key (pk, ck))"
+TABLE_DEF_MASTER = TABLE_DEF_BASE + " with cdc = {'enabled': true}"
+
+class CDCReplicationTest(ClusterTester):
+    def test_replication(self):
+        print("Adding node to db_cluster... ")
+        node2 = self.db_cluster.add_nodes(count=1, enable_auto_bootstrap=True)[0]
+        print("Waiting for added node to initialize...")
+        self.db_cluster.wait_for_init(node_list=[node2])
+
+        print("Adding node to db_cluster... ")
+        node3 = self.db_cluster.add_nodes(count=1, enable_auto_bootstrap=True)[0]
+        print("Waiting for added node to initialize...")
+        self.db_cluster.wait_for_init(node_list=[node3])
```

Comment:
why don't you create the db_cluster with initial size of 3?

Reply:
Because SCT starts the nodes with `auto_bootstrap=off` (https://github.com/scylladb/scylla-cluster-tests/issues/2247), and CDC won't work properly if the nodes aren't bootstrapped properly.

Reply:
If this really the case and we must start empty cluster with bootstrap = true, we will change that.
No need to handle it specifically in the test.

Reply:
It should at least be configurable.

Reply:
it's should default to True, and seem like we are failing to do so.
and on initial node boot it's False, and only later we change it to True.

I thought we already cleared that in SCT code, I was wrong...